### PR TITLE
Tg 12 tasks

### PR DESCRIPTION
**Pull request by ProDIGGY-vinilos**

### CODE REVIEW

File: src/controllers/tasks.js [hunk 1]
```diff
@@ -1,112 +1,60 @@
...
-});
-
-router.get('/:id', (req, res) => {
-  const findTask = tasks.find((task) => task.id === parseInt(req.params.id, 10));
-  if (findTask) {
-    res.status(200).json({
-      data: findTask,
+const createTask = async (req, res) => {
+  try {
+    const task = new Tasks({
```
Comment (agustin-schiaffino) [inline src/controllers/tasks.js:5]: I would rename this to newTask

File: src/controllers/tasks.js [hunk 2]
```diff
@@ -1,112 +1,60 @@
...
-        });
-      } else {
-        res.status(200).json({
-          message: 'task created',
-        });
-      }
+  } catch (error) {
+    return res.status(400).json({
+      message: `Cannot create Task: ${error}`,
+      error: true,
```
Comment (agustin-schiaffino) [inline src/controllers/tasks.js:17]: please remember to change the status code to 500 on every catch, and also you are missing to add data: undefined as the team agreed on

File: src/controllers/tasks.js [hunk 3]
```diff
@@ -1,112 +1,60 @@
...
+      error: false,
     });
-  }
-  if (findTaskId === undefined) {
-    return res.status(404).json({
-      message: 'task not found',
+  } catch (error) {
+    return res.status(400).json({
+      message: `Cannot get all Tasks:  ${error}`,
+      error,
```
Comment (agustin-schiaffino) [inline src/controllers/tasks.js:34]: same comment as on line 15 to 17

File: src/controllers/tasks.js [hunk 4]
```diff
@@ -1,112 +1,60 @@
...
+    const task = await Tasks.findById(id);
+    return res.status(200).json({
+      message: `Task with ${id} found successfully`,
+      data: task,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: `Cannot get Task: ${error}`,
+      error: true,
```
Comment (agustin-schiaffino) [inline src/controllers/tasks.js:51]: same comment as on line 15 to 17

**Comment by agustin-schiaffino** [comment]
please, order the methods on the controller file, so that we could keep the consistency: first the getAll method, then the getById and last one the create.

File: src/controllers/tasks.js (hunk 1 above)
Comment (ProDIGGY-vinilos) [reply src/controllers/tasks.js]: Fixed

File: src/controllers/tasks.js (hunk 2 above)
Comment (ProDIGGY-vinilos) [reply src/controllers/tasks.js]: Fixed

File: src/controllers/tasks.js (hunk 3 above)
Comment (ProDIGGY-vinilos) [reply src/controllers/tasks.js]: Fixed

File: src/controllers/tasks.js (hunk 4 above)
Comment (ProDIGGY-vinilos) [reply src/controllers/tasks.js]: Fixed

File: src/controllers/tasks.js [hunk 5]
```diff
@@ -1,112 +1,78 @@
...
-          message: 'An error occurred',
-        });
-      } else {
-        res.status(200).json({
-          message: 'Task updated',
-        });
-      }
+};
+
+const newTask = async (req, res) => {
```
Comment (agustin-schiaffino) [inline src/controllers/tasks.js:54]: here is no need to change the method name, I was refering only to the const on line 56, please check this a revert this name to createTask

File: src/models/Tasks.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const taskSchema = new Schema({
+  description: { type: String },
```
Comment (agustin-schiaffino) [inline src/models/Tasks.js:6]: you should add required: true here since it is the only field of the resource

**Review by agustin-schiaffino — DISMISSED** [review]

File: postman_collection.json [hunk 7]
```diff
@@ -1,10 +1,165 @@
...
+			"name": "localhost:5000/projects",
+			"request": {
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": ""
+				}
+			},
+			"response": []
+		},
```
Comment (agustin-schiaffino) [inline postman_collection.json:162]: I think this lines should be deleted, because projects is already on line 164

**Review by agustin-schiaffino — APPROVED** [review]

**Pull request merged**